src/platformsh.js

```diff
@@ -1,67 +1,447 @@
+     *   If the relationship/index pair requested does not exist.
+     */
+    credentials(relationship, index = 0) {
+        if (!this.isValidPlatform()) {
+            throw new Error('You are not running on Platform.sh, so relationships are not available.');
+        }
+        if (this.inBuild()) {
+            throw new Error('Relationships are not available during the build phase.');
+        }
+        if (! this.relationshipsDef[relationship]) {
+            throw new RangeError(`No relationship defined: ${relationship}.  Check your .platform.app.yaml file.`);
+        }
+        if (! this.relationshipsDef[relationship][index]) {
+            throw new RangeError(`No index ${index} defined for relationship: ${relationship}.  Check your .platform.app.yaml file.`);
+        }
+
+        return this.relationshipsDef[relationship][index];
+    }
+
+    /**
+     * Returns a variable from the VARIABLES array.
+     *
+     * Note: variables prefixed with `env:` can be accessed as normal environment variables.
+     * This method will return such a variable by the name with the prefix still included.
+     * Generally it's better to access those variables directly.
+     *
+     * @param {string} name
+     *   The name of the variable to retrieve.
+     * @param {any} defaultValue
+     *   The default value to return if the variable is not defined. Defaults to null.
+     * @return mixed
+     *   The value of the variable, or the specified default.  This may be a string or an array.
+     */
+    variable(name, defaultValue = null) {
+        if (!this.isValidPlatform()) {
+            return defaultValue;
+        }
+
+        return this.variablesDef[name] || defaultValue;
+    }
+
+    /**
+     * Returns the full variables array.
+     *
+     * If you're looking for a specific variable, the variable() method is a more robust option.
+     * This method is for cases where you want to scan the whole variables list looking for a pattern.
+     *
+     * @return {object}
+     *   The full variables definition.
+     */
+    variables() {
+        if (!this.isValidPlatform()) {
+            throw new Error('You are not running on Platform.sh, so the variables array is not available.');
+        }
+
+        return this.variablesDef;
+    }
+
+    /**
+     * Returns the application definition object.
+     *
+     * This is, approximately, the .platform.app.yaml file as a nested array.  However, it also
+     * has other information added by Platform.sh as part of the build and deploy process.
+     *
+     * @return {object}
+     *   The application definition object.
+     */
+    application() {
+        if (!this.isValidPlatform()) {
+            throw new Error('You are not running on Platform.sh, so the application definition is not available.');
+        }
+
+        return this.applicationDef;
+    }
+
+    /**
+     * The absolute path to the application directory.
+     *
+     * @returns {string}
+     */
+    get appDir() {
```

Comment:
why `appDir` and not `applicationDirectory`?

Reply:
Because the environment variable is PLATFORM_APP_DIR.  For the most part we tried to stick with the underlying names for consistency.